src/test/java/com/stormeye/evaluation/BlockStepDefinitions.java

```diff
@@ -215,25 +215,33 @@ public void theBodyOfTheReturnedBlockIsEqualToTheBodyOfTheReturnedTestNodeBlock(
 
         logger.info("Then the body of the returned block is equal to the body of the returned test node block");
 
-        final JsonBlockData latestBlockSdk = contextMap.get("blockDataSdk");
+        JsonBlockData latestBlockSdk = contextMap.get("blockDataSdk");
         final JsonNode latestBlockNode = mapper.readTree(contextMap.get("blockDataNode").toString());
 
+        if (!latestBlockSdk.getBlock().getHash().toString().equals(latestBlockNode.get("hash").asText())){
+            //Fixes intermittent syncing issues with nctl/sdk latest blocks
+            latestBlockSdk = getCasperService().getBlock();
+            contextMap.put("blockDataSdk", latestBlockSdk);
+        }
+
         assertThat(latestBlockSdk.getBlock().getBody(), is(notNullValue()));
 
         assertThat(latestBlockSdk.getBlock().getBody().getProposer().toString().equals(latestBlockNode.get("body").get("proposer").asText()), is(true));
```

Comment:
Use correct matcher for assertion here too.
`assertThat(latestBlockSdk.getBlock().getBody().getProposer().toString(), is(latestBlockNode.get("body").get("proposer").asText()))`